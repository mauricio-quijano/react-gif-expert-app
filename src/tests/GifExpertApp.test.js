import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Tests on <GifExpertApp />', () => {
    test('should display component correctly', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should display a list of categories', () => {
        const categories = ['Dragon Ball', 'Hunter x Hunter'];
        const wrapper = shallow(
            <GifExpertApp defaultCategories={categories} />
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});
