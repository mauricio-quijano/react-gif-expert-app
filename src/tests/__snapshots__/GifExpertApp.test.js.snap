// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifExpertApp /> should display a list of categories 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="Hunter x Hunter"
      key="Hunter x Hunter"
    />
  </ol>
</Fragment>
`;

exports[`Tests on <GifExpertApp /> should display component correctly 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
